### PR TITLE
PTX-13848: Improve tagging logic for reporting to dashboard

### PR DESCRIPTION
**Pull request by CharudathGopal**
- Changed tagging logic
- Added code to push TestRail ID to Aetos
- Minor nit fixes

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -403,18 +395,11 @@ func poolResizeIsInProgress(poolToBeResized *api.StoragePool) bool {
...
 			updatedPoolToBeResized := pools[poolToBeResized.Uuid]
-
 			dash.VerifyFatal(updatedPoolToBeResized != nil, true, "Validate pool to be resized exist")
-			dash.VerifyFatal(waitCount, 0, "timed out waiting for pool resize to finish")
 			if updatedPoolToBeResized.LastOperation.Status != api.SdkStoragePool_OPERATION_SUCCESSFUL {
 				dash.VerifyFatal(updatedPoolToBeResized.LastOperation.Status != api.SdkStoragePool_OPERATION_FAILED, true, fmt.Sprintf("PoolResize has failed. Error: %s", updatedPoolToBeResized.LastOperation))
-				err = ValidatePoolRebalance()
-				if err != nil {
-					dash.VerifyFatal(err, nil, "Error while pool rebalance")
-				}
```
Comment (mborodin-px) [inline tests/basic/storage_pool_test.go:414]: looks like something wrong here, as you remove some steps from the test, rebase issue?

**Review by lavong599 — APPROVED** [review]
Fairly simple changes.  Don't see anything that needs attention.

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -193,10 +191,7 @@ var _ = Describe("{StoragePoolExpandDiskAdd}", func() {
 })
 
 var _ = Describe("{PoolResizeDiskReboot}", func() {
-	JustBeforeEach(func() {
-		StartTorpedoTest("PoolResizeDiskReboot", "Initiate pool expansion using resize-disk and reboot node", nil)
-	})
-
+	StartTorpedoTest("PoolResizeDiskReboot", "Initiate pool expansion using resize-disk and reboot node", nil)
```
Comment (mborodin-px) [inline tests/basic/storage_pool_test.go:194]: StartTorpedoTest mehod takes 4 arguments, but you pass only 3

File: tests/common.go [hunk 3]
```diff
@@ -4854,10 +4869,10 @@ func updatePxRuntimeOpts() error {
 }
 
 //StartTorpedoTest starts the logging for torpedo test
-func StartTorpedoTest(testName, testDescription string, tags []string) {
+func StartTorpedoTest(testName, testDescription string, tags map[string]string, testRepoID int) {
```
Comment (mborodin-px) [inline tests/common.go:4872]: what is the testRepoId? When you invoke this method you pass `testrailID`, so maybe it is better to name it as `testrailID`?

File: tests/common.go [hunk 4]
```diff
@@ -4854,10 +4869,10 @@ func updatePxRuntimeOpts() error {
...
 	TestLogger = CreateLogger(fmt.Sprintf("%s.log", testName))
 	SetTorpedoFileOutput(log, TestLogger)
-	dash.TestCaseBegin(testName, testDescription, "", tags)
+	dash.TestCaseBegin(testName, testDescription, strconv.Itoa(testRepoID), tags)
```
Comment (mborodin-px) [inline tests/common.go:4875]: why do we need to make the conversion here, maybe we can pass that variable as a string?

Comment (CharudathGopal) [reply tests/common.go:4875]: testRailID is declared and used as int all our the place, which is correct. But Aetos API expects this to be a string. So making the conversion.

**Comment by CharudathGopal** [comment]
@mborodin-px Here are the successful run with image built from the PR.

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Aniket/job/tp-basic-sharedv4/lastCompletedBuild/

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**